backend/app/training/scrape_products.py

```
@@ -251,18 +252,54 @@ async def scrape_products(sitemap_url, scrape_limit=float('inf'), sample_size=5,
+    sitemap_url: Optional[str] = None,
+    scrape_limit: int = 5,
+    sample_size: int = 3,
+    max_concurrent: int = 20,
+) :
+    """
+    Scrape products by first trying an auto-discovered sitemap.
+    If no products are found there and a user-provided sitemap_url exists, retry using that.
+    Returns the first non-empty product list or [] if none yield results.
+    """
+    # Helper to attempt scraping a given sitemap
+    async def _attempt(sitemap: str):
```

Ruff (0.8.2)

Missing return type annotation for private function `_attempt`

(ANN202)